presto-docs/src/main/sphinx/spill/overview.rst

```diff
@@ -0,0 +1,67 @@
+Spill to disk
+=============
+
+.. contents::
+    :local:
+    :backlinks: none
+    :depth: 1
+
+Overview
+--------
+
+In the case of memory intensive operations, Presto features offloading
+intermediate operator results to disk. The goal of this mechanism is to
+enable executing queries requiring amounts of memory exceeding per query
+or per node limits.
+
+The mechanism is similar to OS level page swapping. However, it is
+reimplemented on the application level to address specific needs of Presto.
+
+Properties related to spilling are described in properties chapter
+:ref:`spill properties<tuning-spill>`.
+
+Revocable memory management
+---------------------------
+
+By default, Presto manages memory needed by drivers manually to guarantee
+fairness in allocating memory to queries and prevent saturation of memory by a
+single query (could result in failure of other queries). In the basic case,
+if the memory requested by a driver exceeds session properties
+`query_max_memory` or `query_max_memory_per_node` the query is killed. This
+strategy is efficient when there is a lot of small queries in the cluster, but
+leads to killing large queries that don't stay within the limits.
+
+To overcome this inefficiency the concept of revocable memory was introduced.
+The driver can request for memory that does not count toward the limits, but it
+has to be capable of returning it back immediately at any time. To prevent
+losing intermediate results, the driver ordered to revoke memory saves a part of
+the results on disk to finish processing it later.
+
+Operator implementations
+------------------------
+
+Details of the revocation handling and returning to processing after being
+revoked are operator dependent. Currently, the mechanism is implemented for the
+following operators.
+
+Joins
+^^^^^
+
+When the spill is enabled, join operator uses revocable memory to store
+hashtable of build table. When requested to revoke memory, the operator saves
+partitions of build table to disk until enough amount of memory can be revoked.
+Iteration over probe table rows is continued for build table rows that are still
+in memory while the other rows are saved to disk for further processing. The
+rows are saved in partitions that correspond to build table partitions.
+
+After the memory is available again for the driver, it iterates over spilled
+partitions of build table. For each partition, it loads corresponding rows of
+probe table from disk to perform joins between those parts.
+
+Aggregations
+^^^^^^^^^^^^
+
+The case of aggregation is similar to join but following differences:
```

Comment:
I would drop statement that aggregations are similar to joins as they are not really.
I think it would be valuable to draft how in high level the mechanism works for joins.
That we are cumulating the final aggregation result in memory. But if we run out of memory for query processing we 
dump the cumulated aggregation states to disk.
And then in the end we merge files spilled to disk in little, constant memory.